x-pack/elastic-agent/pkg/agent/install/install_unix.go

```diff
@@ -7,8 +7,59 @@
 
 package install
 
+import (
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/application/paths"
+)
+
 // postInstall performs post installation for unix-based systems.
 func postInstall() error {
 	// do nothing
 	return nil
 }
+
+func checkPackageInstall() bool {
+	if runtime.GOOS != "linux" {
+		return false
+	}
+
+	// NOTE searching for english words might not be a great idea as far as portability goes.
+	// list all installed packages then search for paths.BinaryName?
+	// dpkg is strange as the remove and purge processes leads to the package bing isted after a remove, but not after a purge
```

Comment:
would like feedback for this suggestion

Reply:
This is expected because remove does not remove modified configuration files, those will remain. Only remove with purge completely removes all data on the system related to a package.

Reply:
Sadly yes, this is expected.

Reply:
i'll keep the comments here as the extra documentation can help if anyone else needs to work on this